### PR TITLE
Added support for tooltip documentation on parameters and type parameters

### LINKED ISSUE #41: Documentation should include remarks and other fields
I'm looking at the documentation of [`PipeReader.AdvanceTo` from the package System.IO.Pipelines](https://www.fuget.org/packages/System.IO.Pipelines/4.5.0/netstandard2.0/lib/System.IO.Pipelines.dll/System.IO.Pipelines/PipeReader#AdvanceTo). To understand that method, it's important to know what is the meaning of its parameters. And while [this is explained in `<remarks>` and `<param>`s](https://github.com/dotnet/corefx/blob/6e44392/src/System.IO.Pipelines/src/System/IO/Pipelines/PipeReader.cs#L39-L48), FuGet does not show them:

I think this should be improved.

### PR DESCRIPTION
**Pull request by TylerBrinkley**
Added support for tooltip documentation on parameters and type parameters.

Closes #41

### CODE REVIEW

**Comment by TylerBrinkley** [comment]
Note, generic references will display as an example ``Dictionary`2``

File: Data/CecilExtensions.cs [hunk 1]
```diff
@@ -306,7 +306,7 @@ public static void WritePrototypeHtml (this MethodDefinition member, TextWriter
             foreach (var p in member.Parameters) {
                 w.Write (head);
                 WriteReferenceHtml (p.ParameterType, w, framework, p.IsOut, isExtensionMethod);
-                w.Write (" <span class=\"c-ar\">");
+                w.Write ($" <span class=\"c-ar\" title=\"{(docs != null && docs.ParametersText.TryGetValue(p.Name, out var paramText) ? paramText : string.Empty)}\">");
```
Comment (praeclarum) [inline Data/CecilExtensions.cs]: This string still needs HTML encoding. It could, for example, contain a "

File: Data/CecilExtensions.cs [hunk 2]
```diff
@@ -323,13 +323,15 @@ public static void WritePrototypeHtml (this MethodDefinition member, TextWriter
...
 
-        private static void WriteGenericParameterListHtml (Collection<GenericParameter> genericParameters, TextWriter w, PackageTargetFramework framework)
+        private static void WriteGenericParameterListHtml (Collection<GenericParameter> genericParameters, TextWriter w, PackageTargetFramework framework, MemberXmlDocs docs)
         {
             w.Write ("&lt;");
             var head = "";
             foreach (var p in genericParameters) {
                 w.Write (head);
-                WriteReferenceHtml (p, w, framework);
+                w.Write ($"<span class=\"c-tr\" title=\"{(docs != null && docs.TypeParametersText.TryGetValue (p.Name, out var paramText) ? paramText : string.Empty)}\">");
```
Comment (praeclarum) [inline Data/CecilExtensions.cs]: Need to escape the "

File: Data/CecilExtensions.cs [hunk 3]
```diff
@@ -412,7 +414,7 @@ public static void WritePrototypeHtml (this PropertyDefinition member, TextWrite
                 foreach (var p in member.GetMethod.Parameters) {
                     w.Write (head);
                     WriteReferenceHtml (p.ParameterType, w, framework);
-                    w.Write (" <span class=\"c-ar\">");
+                    w.Write ($" <span class=\"c-ar\" title=\"{(docs != null && docs.ParametersText.TryGetValue (p.Name, out var paramText) ? paramText : string.Empty)}\">");
```
Comment (praeclarum) [inline Data/CecilExtensions.cs]: Encode the text

**Comment by TylerBrinkley** [comment]
I've added encoding.

**Comment by praeclarum** [comment]
Thank you!

**Pull request merged**